src/noForInRule.ts

```diff
@@ -25,16 +25,21 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithWalker(new NoForInRuleWalker(sourceFile, this.getOptions()));
+        return this.applyWithFunction(sourceFile, walk);
     }
 }
 
-class NoForInRuleWalker extends Lint.RuleWalker {
-    protected visitForInStatement(node: ts.ForInStatement): void {
-        const initializer: string = node.initializer.getText();
-        const expression: string = node.expression.getText();
+function walk(ctx: Lint.WalkContext<void>) {
+    function cb(node: ts.Node): void {
+        if (node.kind === ts.SyntaxKind.ForInStatement) {
```

Comment:
> `node.kind === ts.SyntaxKind.ForInStatement`

Use `tsutils.isForInStatement(node)`. It returns `node is ts.ForInStatement` as a hint to the type checker so `node` will be narrowed to that type in the for loop. That way you won't have to cast `(<ts.ForInStatement>node)` in there.